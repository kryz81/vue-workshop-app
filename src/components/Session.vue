<template>
  <div>
    <div v-if="loading">Loading session...</div>
    <div v-if="session">
      <h4 class="mb-4">{{ session.title }}</h4>
      <p>{{ session.description }}</p>
    </div>
  </div>
</template>

<script>
import { getSessionById } from "../services/sessions";

export default {
  props: {
    id: {
      type: String,
      required: true
    }
  },
  data: () => ({
    session: undefined,
    loading: false
  }),
  async mounted() {
    try {
      this.loading = true;
      const res = await getSessionById(this.id);
      this.session = res.data;
    } catch (err) {
      console.log(err);
    } finally {
      this.loading = false;
    }
  }
};
</script>
