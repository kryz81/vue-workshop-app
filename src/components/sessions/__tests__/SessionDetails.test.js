import { mount } from "@vue/test-utils";
import flushPromises from "flush-promises";
import Session from "../SessionDetails";

jest.mock("../../../services/sessions");

describe("Session", () => {
  it("renders correctly", async () => {
    // when
    const wrapper = mount(Session, {
      stubs: ["router-view", "router-link"],
      propsData: { id: "12345" }
    });

    await flushPromises();

    // then
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.props("id")).toEqual("12345");
  });
});
