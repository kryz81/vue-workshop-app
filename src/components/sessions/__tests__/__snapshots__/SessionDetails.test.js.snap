// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session renders correctly 1`] = `
<div>
  <div>Loading session...</div>
  <div>
    <h4 class="mb-2">Node-Tools f√ºr Frontend-Entwickler &amp; Webpack</h4>
    <div class="row">
      <div class="col-12">
        <p class="mb-4">Robert Weber</p>
        <div class="mb-4"><span class="badge badge-secondary mr-2">ROOM 4a</span> <span class="badge badge-secondary mr-2">DAY 2</span> <span class="badge badge-primary">9:00 - 12:30</span></div>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
        <p class="mt-5"><strong><router-link-stub to="[object Object]"><svg width="1em" height="1em" viewBox="0 0 20 20" focusable="false" role="img" alt="icon" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-star-half b-icon bi"><g><path fill-rule="evenodd" d="M7.354 7.119l2.184-4.327A.516.516 0 0110 2.5c.183 0 .366.097.465.292l2.184 4.327 4.898.696A.537.537 0 0118 8.32a.55.55 0 01-.17.445l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L10 15.187l-4.389 2.256a.52.52 0 01-.146.05c-.341.06-.668-.254-.6-.642l.83-4.73-3.522-3.356a.55.55 0 01-.172-.403.59.59 0 01.084-.302.513.513 0 01.37-.245l4.898-.696zM10 14.027c.08 0 .16.018.232.056l3.686 1.894-.694-3.957a.564.564 0 01.163-.505l2.907-2.77-4.053-.576a.525.525 0 01-.393-.288l-1.847-3.658-.001.003v9.8z" clip-rule="evenodd"></path></g></svg> Write a review</router-link-stub></strong></p>
      </div>
      <!---->
    </div>
    <router-view-stub name="session" sessionid="12345"></router-view-stub>
  </div>
</div>
`;
